docs/script-console-scripts/configure-yadocker-cloud.groovy

```diff
@@ -0,0 +1,417 @@
+        nodeProperties << (new EnvironmentVariablesNodeProperty(envEntries))
+    }
+    if(obj.optJSONObject('tool_locations')) {
+        HashMap<String,String> tool = obj.optJSONObject('tool_locations') as HashMap<String,String>
+        List<ToolLocationNodeProperty.ToolLocation> toolLocations = [] as List<ToolLocationNodeProperty.ToolLocation>
+        (tool.keySet() as String[]).each { location ->
+            //tool location is only valid if it contains a type i.e. @ symbol
+            if(location.contains('@') && detectGlobalToolExists(location)) {
+                toolLocations << (new ToolLocationNodeProperty.ToolLocation(location, tool[location]))
+            }
+            else {
+                //alert the user they configured an invalid tool type or name in tool_locations
+                println "WARNING: Invalid tool: '${location}'.  Format should be 'type@name' where both type and name already exist in global tool configurations."
+            }
+        }
+        nodeProperties << (new ToolLocationNodeProperty(toolLocations))
+    }
+    //set NODE PROPERTIES
+    if(nodeProperties) {
+        dockerSlaveTemplate.setNodeProperties(nodeProperties)
+    }
+    return dockerSlaveTemplate
+}
+
+def bindJSONToList(Class type, Object src) {
+    if(!(type == DockerCloud) && !(type == DockerSlaveTemplate)) {
+        throw new Exception("Must use DockerCloud or DockerSlaveTemplate class.")
+    }
+    //docker_array should be a DockerCloud or DockerSlaveTemplate
+    ArrayList<?> docker_array
+    if(type == DockerCloud){
+        docker_array = new ArrayList<DockerCloud>()
+    }
+    else {
+        docker_array = new ArrayList<DockerSlaveTemplate>()
+    }
+    //cast the configuration object to a Docker instance which Jenkins will use in configuration
+    if (src instanceof JSONObject) {
+        //uses string interpolation to call a method
+        //e.g instead of newDockerCloud(src) we use instead...
+        docker_array.add("new${type.getSimpleName()}"(src))
+    }
+    else if (src instanceof JSONArray) {
+        for (Object o : src) {
+            if (o instanceof JSONObject) {
+                docker_array.add("new${type.getSimpleName()}"(o))
+            }
+        }
+    }
+    return docker_array
+}
+
+if(!Jenkins.instance.isQuietingDown()) {
+    ArrayList<DockerCloud> clouds = new ArrayList<DockerCloud>()
+    clouds = bindJSONToList(DockerCloud.class, clouds_yadocker)
+    if(clouds.size() > 0) {
+        dockerConfigUpdated = true
+        Jenkins.instance.clouds.removeAll(DockerCloud)
+        Jenkins.instance.clouds.addAll(clouds)
+        clouds*.name.each { cloudName ->
+            println "Configured Yet Another Docker cloud ${cloudName}"
+        }
+        Jenkins.instance.save()
+    }
+    else {
+        println 'Nothing changed.  No Yet Another docker clouds to configure.'
+    }
+}
+else {
+    println 'Shutdown mode enabled.  Configure Yet Another Docker clouds SKIPPED.'
+}
+
+null
```

Comment:
I updated with why I have `null` at the end.  It's explained in a comment.  I prefer the script to return `null` rather than print an empty line which affects script console output.